modin/core/execution/dask/common/engine_wrapper.py

```diff
@@ -13,36 +13,46 @@
 
 """Module houses class responsible for execution of remote operations."""
 
+from typing import Tuple
+from distributed import Future
 from distributed.client import default_client
+from modin.utils import Invocable
 
 
 class DaskWrapper:
     """The class responsible for execution of remote operations."""
 
     @classmethod
-    def deploy(cls, func, *args, num_returns=1, pure=None, **kwargs):
+    def deploy(
+        cls,
+        invocable,
+        num_returns=1,
+        pure=True,
+    ):
         """
         Deploy a function in a worker process.
 
         Parameters
         ----------
-        func : callable
-            Function to be deployed in a worker process.
-        *args : list
-            Additional positional arguments to be passed in `func`.
+        invocable : Invocable
+            The function with its args and kwargs to be deployed in a worker process.
+            In order to avoid spawning a worker process from within another worker process,
+            this invocable must have a callable func rather than a future of a function.
         num_returns : int, default: 1
             The number of returned objects.
-        pure : bool, optional
+        pure : bool, default: True
```

Comment:
`pure=True` is the default behavior of Dask ([see here](https://docs.dask.org/en/stable/futures.html#distributed.Client.submit)); making this the case in our codebase as well may help avoid potential confusion in the future.